src/Categories/Functor/Bialgebra.agda

```diff
@@ -0,0 +1,66 @@
+{-# OPTIONS --without-K --safe #-}
+
+module Categories.Functor.Bialgebra where
+
+open import Level
+open import Categories.Category using (Category)
+open import Categories.Functor
+open import Categories.Functor.Algebra
+open import Categories.Functor.Coalgebra
+open import Categories.NaturalTransformation
+
+{-
+given two endofunctors T and F on a category C, we can define a (T,F)-bialgebra to be an object of C equipped with the structure of a T-algebra and an F-coalgebra, i.e. just a triple (A,a,c), where a:TA→A and c:A→FA.
+μ is a distributive law (of T over F): a natural transformation μ:TF→FT.
+A μ-bialgebra is a (T,F)-bialgebra (A,a,c) such that c∘a=Fa∘μ_A∘Tc
+[category theory - How are F-bialgebras defined? - Mathematics Stack Exchange](https://math.stackexchange.com/questions/3057781/how-are-f-bialgebras-defined#answer-3057859)
+-}
+
+private
+  variable
+    o ℓ e : Level
+
+module _ {C : Category o ℓ e} where
+
+  module _ (T : Endofunctor C) (F : Endofunctor C) where
+    Distr-law : Set (o ⊔ ℓ ⊔ e)
+    Distr-law = NaturalTransformation (T ∘F F) (F ∘F T)
+
+  record μ-Bialgebra (T : Endofunctor C) (F : Endofunctor C) (μ : Distr-law T F)
+       : Set (o ⊔ ℓ ⊔ e) where
+    open Category C
+    open Functor
+    open NaturalTransformation
+    field
+      A : Obj
+      a₁ : F-Algebra-on T A
+      c₁ : F-Coalgebra-on F A
+    module F = Functor F
+    module T = Functor T
+    field
+      respects-μ : c₁ ∘ a₁ ≈ ((F.₁) a₁) ∘ (μ .η A) ∘ ((T.₁) c₁)
+    --to access the (co)algebras induced by the morphisms:
+    a : F-Algebra T
```

Comment:
The problem is: a bialgebra morphism is a C-morphism which _is_ a T-Algebra morphism _and_ an F-Coalg morphism. I use these accessors at line 59 &60 to define the commutation properties for bialgebra morphisms. The reason I needed to factor `is-F-Algebra-Morphism` out of `F-Algebra-Morphism` and `F-Algebra-on` out of `F-Algebra` respectively is because the latter are essentially the former with the object(s) hidden under an existential.
What I _could_ do is define helpers that "package" e.g. `f0: F-Algebra-on A0` into its existential `record{f=f0; A = A0} : F-Algebra`, which I could define in `F-Algebra`. I'm not sure the conversion can be elided entirely though.

Reply:
Aren't `a` and `c` in the record module `μ-Bialgebra` though? so they shouldn't escape into the global namespace. I added them because for the definition we need to pass `a₁` and `c₁`, but for usability you will usually want to access the (co)algebras induced by them. They are meant to be used in a way akin to the `₀ = F₀;  ₁ = F₁` shorthands in the `Functor` record module.

Reply:
`a` and `c` will escape if your open this record.  And those are short, not-really-meaningful names. Could they be defined locally at the use site instead?

Functors are pervasive, so using a few shorthands (that are not likely to already be used in user code) makes sense. And it took a lot of discussion to settle on these.

Reply:
I renamed them to `alg` & `coalg` respectively in b116824298963c953a67f3145d8dd5ad1f48a83f,  is that fine?

Reply:
excellent.